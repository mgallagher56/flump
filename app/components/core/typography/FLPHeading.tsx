import type { PropsWithChildren } from 'react';
import React from 'react';

import type { HeadingProps } from '@chakra-ui/react';
import { Heading } from '@chakra-ui/react';

const FLPHeading: React.FC<PropsWithChildren<HeadingProps>> = (props) => {
  return <Heading {...props}>{props.children}</Heading>;
};

export default FLPHeading;
