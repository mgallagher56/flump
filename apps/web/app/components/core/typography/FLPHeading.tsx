import type { FC, PropsWithChildren } from 'react';

import type { HeadingProps } from '@chakra-ui/react';
import { Heading } from '@chakra-ui/react';

const FLPHeading: FC<PropsWithChildren<HeadingProps>> = ({ color = 'blue.500', ...props }) => {
  return (
    <Heading color={color} {...props}>
      {props.children}
    </Heading>
  );
};

export default FLPHeading;
