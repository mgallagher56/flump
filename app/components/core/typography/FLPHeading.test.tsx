import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import FLPHeading from './FLPHeading';

const sampleText =
  'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.';

describe('FLPHeading', () => {
  test('renders as expected', () => {
    const { container } = render(<FLPHeading as="h1">{sampleText}</FLPHeading>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a size', () => {
    const { container } = render(<FLPHeading size="lg">{sampleText}</FLPHeading>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a color', () => {
    const { container } = render(<FLPHeading colorScheme="blue">{sampleText}</FLPHeading>);
    expect(container).toMatchSnapshot();
  });
  test('render only number of lines', () => {
    const { container } = render(
      <FLPHeading as="h1" size="4xl" noOfLines={1}>
        {sampleText}
      </FLPHeading>
    );
    expect(container).toMatchSnapshot();
  });
});
