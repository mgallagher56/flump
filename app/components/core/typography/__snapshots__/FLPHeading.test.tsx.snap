// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPHeading > render only number of lines 1`] = `
<div>
  <h1
    class="chakra-heading css-5bia5c"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h1>
</div>
`;

exports[`FLPHeading > renders as expected 1`] = `
<div>
  <h1
    class="chakra-heading css-1e3tj"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h1>
</div>
`;

exports[`FLPHeading > renders as expected with a color 1`] = `
<div>
  <h2
    class="chakra-heading css-1e3tj"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h2>
</div>
`;

exports[`FLPHeading > renders as expected with a size 1`] = `
<div>
  <h2
    class="chakra-heading css-1e3tj"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h2>
</div>
`;
